src/main/java/study/step4/Racing.java

```diff
@@ -0,0 +1,60 @@
+package study.step4;
+
+import static java.util.stream.Collectors.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import study.step4.model.Car;
+import study.step4.model.Movement;
+import study.step4.ui.ResultView;
+
+/**
+ * Racing Game Class
+ *
+ * 자동차 경주 클래스
+ *
+ */
+public class Racing {
+    private List<Car> carList;
+    private Movement movement;
+    private ResultView resultView;
+
+    public Racing() {
+        carList = new ArrayList<>();
+        movement = new Movement();
+        resultView = new ResultView();
+    }
+
+    public void setUp(String[] names) {
+        IntStream.range(0, names.length).forEach(i -> carList.add(new Car(names[i])));
+    }
+
+    public List<Car> getCarList() {
+        return this.carList;
+    }
+
+    public void run() {
+        carList.forEach(car -> car.move(movement.stopAndMove()));
+    }
+
+    public void printRacingResult(String printToken) {
+        resultView.printRacingResult(carList, printToken);
+    }
+
+    public void printRacingWinner(String message) {
+        resultView.printRacingWinner(this.getWinners(), message);
+    }
+
+    private List<Car> getWinners() {
+        int winnerPosition = carList.stream()
+            .mapToInt(Car::getPosition)
+            .max()
+            .orElse(-1);
+
+        return carList.stream()
+            .filter(car -> car.getPosition() == winnerPosition)
+            .collect(toList());
+    }
```

Comment:
작성하신 코드를 stream을 사용하지 않고 작성하게되면 아래와 같이 표현됩니다.
for문 안에 if로 인해 depth가 2depth가 되었습니다. 프로그래밍 요구사항을 확인해보시면 **indent(인덴트, 들여쓰기) depth를 2를 넘지 않도록 구현한다. 1까지만 허용한다 라고 되어있습니다.** 전달드린 부분에 대해서 분리 방법을 고민해봐 주세요 :-)
```suggestion
    private List<Car> getWinners() {
        boolean seen = false;
        int best = 0;
        for (Car car1 : carList) {
            int position = car1.getPosition();
            if (!seen || position > best) {
                seen = true;
                best = position;
            }
        }
        int winnerPosition = seen ? best : -1;

        List<Car> list = new ArrayList<>();
        for (Car car : carList) {
            if (car.getPosition() == winnerPosition) {
                list.add(car);
            }
        }
        return list;
    }
```

Reply:
넵 감사합니다!